src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelper.cs

```diff
@@ -308,19 +361,21 @@ private StringBuilder GetStringBuilder()
         /// <returns>The generated URL.</returns>
         protected virtual string GenerateUrl(string protocol, string host, VirtualPathData pathData, string fragment)
         {
-            if (pathData == null)
+            return GenerateUrl(protocol, host, pathData?.VirtualPath, fragment);
```

Comment:
If users have overriden this method, then they would see it not getting called when using Dispatcher. IMO this is fine as we have said we need not have 100% back compat.

Reply:
It might also be a good reason to make a different implementation.

If a user replaced UrlHelper, then they also wrote a custom factory. Therefore they don't get any of this for free.